### PR TITLE
Ignore errors that have Fn::Transform in the path

### LINKED ISSUE #2008: Duplicate properties on a resource can cause unhandled exceptions
*cfn-lint version: (`cfn-lint --version`)*
0.48.3

*Description of issue.*
Duplicate properties can cause unhandled exceptions

Please provide as much information as possible:
* Template linting issues:
  * Please provide a CloudFormation sample that generated the issue.
  * If present, please add links to the (official) documentation for clarification.
  * Validate if the issue still exists with the latest version of `cfn-lint` and/or the latest Spec files

e.g.
  PublicHttpsListener:
    Type: 'AWS::ElasticLoadBalancingV2::Listener'
    Properties:
      LoadBalancerArn: !Ref PublicLoadBalancer
      Port: 443
      Protocol: HTTPS
      Certificates: !Ref LoadBalancerCertificateArn
      DefaultActions:
      - Type: fixed-response
        FixedResponseConfig:
          StatusCode: 404
      Port: 443
      Protocol: HTTPS

results in:

Traceback (most recent call last):
  File "/Users/nick.chadwick/Projects/msd/miniconda3/bin/cfn-lint", line 10, in <module>
    sys.exit(main())
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/cfnlint/__main__.py", line 28, in main
    (template, rules, errors) = cfnlint.core.get_template_rules(filename, args)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/cfnlint/core.py", line 187, in get_template_rules
    (template, errors) = cfnlint.decode.decode(filename)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/cfnlint/decode/__init__.py", line 28, in decode
    template = cfn_yaml.load(filename)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/cfnlint/decode/cfn_yaml.py", line 237, in load
    return loads(content, filename)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/cfnlint/decode/cfn_yaml.py", line 215, in loads
    template = loader.get_single_data()
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/yaml/constructor.py", line 51, in get_single_data
    return self.construct_document(node)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/yaml/constructor.py", line 55, in construct_document
    data = self.construct_object(node)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/yaml/constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/cfnlint/decode/cfn_yaml.py", line 81, in construct_yaml_map
    value = self.construct_object(value_node, False)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/yaml/constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/cfnlint/decode/cfn_yaml.py", line 81, in construct_yaml_map
    value = self.construct_object(value_node, False)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/yaml/constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/cfnlint/decode/cfn_yaml.py", line 81, in construct_yaml_map
    value = self.construct_object(value_node, False)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/yaml/constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "/Users/nick.chadwick/Projects/msd/miniconda3/lib/python3.7/site-packages/cfnlint/decode/cfn_yaml.py", line 98, in construct_yaml_map
    key, mapping[key].start_mark.line + 1),
AttributeError: 'int' object has no attribute 'start_mark'

* Feature request:
  * Please provide argumentation about the missing feature. Context is key!

Cfn-lint uses the [CloudFormation Resource Specifications](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) as the base to do validation. These files are included as part of the application version. Please update to the latest version of `cfn-lint` or update the spec files manually (`cfn-lint -u`)

### LINKED ISSUE #2016: E7003 Errors when using Fn::Transform inside a Mapping
*cfn-lint version: 0.49.2*

*Description of issue.*
#2006 tightened what is considered valid for use in a Mapping.  This causes it to reject what otherwise appears to be a valid use of `Fn::Transform` as the body of a Mapping.

For example, this snippet is valid CFN:

```yaml
Mappings:
  AwsAgentPlatformMap:
    Fn::Transform:
      Name: AWS::Include
      Parameters:
        Location: s3://my-bucket-name/version/3.0.1/amazonlinux2/a-json-file.json
```

This usage trips the newly enhanced regex:

```
E7003 Mapping key (Fn::Transform) has invalid name. Name has to be alphanumeric, '-' or '.'
```

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:* https://github.com/aws-cloudformation/cfn-lint/issues/476
fix #2016
*Description of changes:*
- Ignore failures if `Fn::Transform` is in the path

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kddejong** [comment]
In relation to #2017

File: src/cfnlint/rules/__init__.py [hunk 1]
```diff
@@ -136,6 +136,8 @@ def wrapper(self, filename, cfn, *args, **kwargs):
 
                 if results:
                     for result in results:
+                        if 'Fn::Transform' in result.path:
```
Comment (PatMyron) [inline src/cfnlint/rules/__init__.py:139]: Assuming we've expanded [short form to long form](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-transform.html#intrinsic-function-reference-transform-declaration) by this point?

Comment (kddejong) [reply src/cfnlint/rules/__init__.py:139]: Good question. Hmm.  Shouldn't the short form be `!Transform`?

**Comment by kddejong** [comment]
may need to add a rule attribute that will specify if we want to include `Fn::Transform` in the errors.

**Comment by andrew-glenn** [comment]
https://github.com/aws-cloudformation/cfn-lint/pull/2173 Is potentially related

**Comment by kddejong** [comment]
Going to close this as it hasn't come up again. Code will remain just in case.

**Pull request closed without merging**